### DISCUSSION TITLE
Daily Feed-in Sensor request

### DISCUSSION THREAD

**Discussion by pepetheprawn** [hultenvp/solis-sensor#115] [General]
Hi, 

Apologies if I'm going about this the wrong way but I'm VERY new to this. I've just got a Solis PV system, and have now got this integrated into HA/HACS and it's all working brilliantly. Thanks so much for everyone's efforts in doing this! The only question I have at the moment is that I'd really like to add how much power I'm sending back to the grid. In the Solis website you can see this with the 'Daily Grid Feed In' parameter but I don't think this exists as a sensor within the Solis integration. Is there any way that it could be added as that would be amazing! Thanks again for all your efforts, I love this stuff! 

-Gordon

**Comment by hultenvp** [comment]
Hey Gordon,

Typically you use HomeAssistant Energy Dash for that, it takes in all available sensors to give you exactly that data.
Here is a screenshot of my energy dash. It combines solar from 2 different inverters and smart meter stats into a clean overview of energy production and consumption. I do assume your problem is you actually miss the grid data that I get from my smart meter.
![image](https://user-images.githubusercontent.com/61835400/167885175-e6ab16b8-6a63-419f-ad58-1441c8a0bd28.png)

That said, the integration does detect grid stats and exposes them as sensors, but the ones offered for readout are "daily grid energy used" and "daily grid energy purchased", not "energy returned". You might be able to get what you want with a [template sensor](https://community.home-assistant.io/t/calculate-sum-two-entities-to-one-sensor/182063) by subtracting energy produced from energy used.
The exact semantics of those sensors seem to vary a bit between models however (mine does not expose them, so I miss the details), so if you see them in your config you should crosscheck against values on m.ginlong.com to ensure they mean what you think they mean. 
If you do not see them, then they are not exposed or Ginlong once more figured out new keys for new models. 
In that case I might be able to add them but I really need your help then as I need some dumps of the communication between the integration and the ginlong server.

**Comment by pepetheprawn** [comment]
Hi, 

Thanks so much for the quick reply. Looking at my energy dashboard I'm missing that link between solar & grid, I expect because I don't have a sensor configured for 'return to grid' because none of the sensors in the list for that (for me) are what I'd expect to see: 
![image](https://user-images.githubusercontent.com/104442544/167898665-17a681a9-4745-48f4-bfcf-41d4512f4d16.png)
If I look at [m.ginlong ](https://m.ginlong.com/main.html) then I can see that there is a value for 'daily grid feed in' that I think would give what I'd like, but I have no idea how to get that into Home Assistant as a sensor: 
![image](https://user-images.githubusercontent.com/104442544/167899082-e9461efd-9b5b-4381-9f4d-9e4a6141ce96.png)
I'm not sure if it helps, but looking at the code for the webpage then I can see it here:
![image](https://user-images.githubusercontent.com/104442544/167902630-4e81e4c7-faf7-47df-a601-f11d78720019.png)
I wouldn't know what's needed to make this appear as a sensor though. 
At the moment I don't yet have a smart meter, that's being installed next week as it happens but it would be good to get the data from the inverter I think as that can see what it's feeding back. If there is any information that I can provide that will help you then please let me know, as I say I'm really new to this and if there's something I can do to help contribute to this brilliant product I'd be happy to do so! 

Thanks again, 

-Gordon

**Reply by hultenvp** [reply]
I get my data from the ginlong backend, which produces (kind of) key-value based responses. In those responses I've never seen Grid Feed, so I can only assume, not having a hybrid inverter, that either grid feed is calculated by the front end or they did extend for newer models.

I see 2 options moving forward:
1. You uncomment the line 273 (latest version) in ginlong_api.py and attach the generated output here. Be careful to anonymize it. If there's grid feed in it I'll add it for you.
>                    #_LOGGER.debug("Payload = %s", payload)
Note that you need to enable debug for solis, see issue [#14](https://github.com/hultenvp/solis-sensor/issues/114#issuecomment-1120263513) how to do that.
2. Not sure if supported, but your new smart meter could be integrated in HA as well and actually gives you the grid data you need, equal to my own setup. 

Cheers

**Reply by pepetheprawn** [reply]
Hi, 
Thanks for the reply. I'll await the install of my smart meter and see if that can give me what I need then. I'll let you know the outcome. 

Thanks again for all your help.

-Gordon

**Reply by viking2010** [reply]
Just to let you know @pepetheprawn, the entity I think you need for the power you've exported to the grid is called "#YourInverter# Daily On-grid Energy". Apply that to the "Return to Grid" section when editing the energy dashboard and you'll get the same display as @hultenvp showed in his screen shot. I'm assuming of course that what you describe as "Grid Feed In" is the amount of energy you've exported.
-Adrian

**Reply by pepetheprawn** [reply]
That's brilliant, I'll give that a go. Thanks for the reply.
